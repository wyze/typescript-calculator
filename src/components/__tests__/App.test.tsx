import { App } from '../App'
import { h } from 'preact'
import snapshot from './utils/snapshot'

const props = {
  actions: {
    add: jest.fn(),
    clear: jest.fn(),
    divide: jest.fn(),
    equals: jest.fn(),
    input: jest.fn(),
    multiply: jest.fn(),
    percent: jest.fn(),
    posneg: jest.fn(),
    subtract: jest.fn(),
  },
  output: '12',
}

describe('<App />', () => {
  it('renders', () => snapshot(<App {...props} />))
})
