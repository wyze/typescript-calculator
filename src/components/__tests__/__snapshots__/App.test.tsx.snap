// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders 1`] = `
.glamor-0,
[data-glamor-0] {
  align-items: center;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  flex-direction: column;
  -webkit-box-align: center;
  -webkit-align-items: center;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
  -webkit-flex-direction: column;
}

<main
  className="glamor-0"
>
  <Hero />
  <Calculator>
    <Display
      output="12"
    />
    <Buttons
      actions={
        Object {
          "add": [Function],
          "clear": [Function],
          "divide": [Function],
          "equals": [Function],
          "input": [Function],
          "multiply": [Function],
          "percent": [Function],
          "posneg": [Function],
          "subtract": [Function],
        }
      }
    />
  </Calculator>
</main>
`;
